packages/core/src/browser/status-bar/status-bar.tsx

```diff
@@ -0,0 +1,192 @@
+        @inject(LabelParser) protected readonly entryService: LabelParser,
+        @inject(FrontendApplicationStateService) protected readonly applicationStateService: FrontendApplicationStateService
+    ) {
+        super();
+        delete this.scrollOptions;
+        this.id = 'theia-statusBar';
+    }
+
+    protected get ready(): Promise<void> {
+        return this.applicationStateService.reachedAnyState('initialized_layout', 'ready');
+    }
+
+    async setElement(id: string, entry: StatusBarEntry): Promise<void> {
+        await this.ready;
+        this.entries.set(id, entry);
+        this.update();
+    }
+
+    async removeElement(id: string): Promise<void> {
+        await this.ready;
+        this.entries.delete(id);
+        this.update();
+    }
+
+    async setBackgroundColor(color?: string): Promise<void> {
+        await this.ready;
+        this.internalSetBackgroundColor(color);
+    }
+
+    protected internalSetBackgroundColor(color?: string): void {
+        this.backgroundColor = color;
+        // tslint:disable-next-line:no-null-keyword
+        this.node.style.backgroundColor = this.backgroundColor ? this.backgroundColor : null;
+    }
+
+    getLayoutData(): StatusBarLayoutData {
+        const entries: StatusBarEntryData[] = [];
+        this.entries.forEach((entry, id) => {
+            entries.push({ id, entry });
+        });
+        return { entries, backgroundColor: this.backgroundColor };
+    }
+
+    setLayoutData(data: StatusBarLayoutData): void {
+        if (data.entries) {
+            data.entries.forEach(entryData => {
+                this.entries.set(entryData.id, entryData.entry);
+            });
+            this.update();
+        }
+        this.internalSetBackgroundColor(data.backgroundColor);
+    }
+
+    protected render(): JSX.Element {
+        const leftEntries: JSX.Element[] = [];
+        const rightEntries: JSX.Element[] = [];
+        const elements = Array.from(this.entries.values()).sort((left, right) => {
+            const lp = left.priority || 0;
+            const rp = right.priority || 0;
+            return rp - lp;
+        });
+        elements.forEach(entry => {
+            if (entry.alignment === StatusBarAlignment.LEFT) {
+                leftEntries.push(this.renderElement(entry));
+            } else {
+                rightEntries.push(this.renderElement(entry));
+            }
+        });
+        const leftElements: React.ReactElement<StatusBarElements> = <StatusBarElements key="statusbar-left-elements" alignment="left" entries={leftEntries} />;
+        const rightElements: React.ReactElement<StatusBarElements> = <StatusBarElements key="statusbar-right-elements" alignment="right" entries={rightEntries} />;
+
+        return <StatusBarView leftElements={leftElements} rightElements={rightElements} />;
+    }
+
+    protected createAttributes(entry: StatusBarEntry): StatusBarEntryAttributes {
```

Comment:
it will produce a new function each time. It would be interesting to try to add https://github.com/palantir/tslint-react, it has useful rules as `jsx-no-bind`, `jsx-no-lamba` to catch such issues in jsx.

It seems that `StatusBarEntryAttributes` is superflous and `createAttributes` can be inlined to `renderElement` as jsx:
```jsx
return <div
    className={entry.command ? 'element hasCommand' : 'element'}
    title={entry.tooltip}
    onClick={() => {...}}
/>
```

after that rules should be able to catch bogus `onClick={() => {...}}`